repository/repository-web/dist/js/controllers/modeshapeDoctorController.js

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright (c) 2020 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * https://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+define(["../init/appController"], function (repositoryControllers) {
+
+    repositoryControllers.controller("modeshapeDoctorController",
+        ["$rootScope", "$scope", "$http", "$uibModal", "dialogConfirm",
+            "dialogPrompt",
+            function ($rootScope, $scope, $http, $uibModal, dialogConfirm,
+                      dialogPrompt) {
+
+                $scope.modeshapeData = null;
+                $scope.modeshapePath = "";
+                $scope.modeshapeWorkspaceId = "";
+
+                $scope.readModeshapeData = function (modeshapeWorkspaceId, modeshapePath) {
+                    $http.get("/rest/namespaces/diagnostics/modeshape/node/" + modeshapeWorkspaceId + "?path=" + modeshapePath)
+                        .then(response => {
+                            $scope.modeshapeData = response.data;
+                            $scope.modeshapePath = modeshapePath;
+                        }).catch(error => {
+                        alert("Error reading modeshape node");
```

Comment:
I think JS alerts are a bit too rough for validation, or anything for that matter. As an alternative, I would propose either "toast-like" warnings, or "validation-like" warning (essentially warnings et al. handled in dynamic divs). There are a few usable examples across the code base.